File: src/homework/Nov26/miniProject/Page/AddProductPage.java
package homework.Nov26.miniProject.Page;

import java.util.Scanner;

import homework.Nov26.miniProject.Product;
import homework.Nov26.miniProject.ProductManager;
import homework.Nov26.miniProject.State;
import homework.Nov26.miniProject.Abstract.Page;

public class AddProductPage extends Page {
	
	public AddProductPage() {}

	@Override
	public void show() {
		ProductManager products = State.getState().getProductManager();
		products.showProducts();
		
		Scanner scanner = State.getState().getScanner();
		int id = products.getSize()+1;
		
		System.out.print("Please enter product name: ");
		String title = scanner.next();
			
		System.out.print("Please enter product price: ");
		long price = scanner.nextLong();
		
		Product newProduct = new Product(id, title, price);
		products.addProduct(newProduct);
		products.showProducts();
		((Page) State.getState().getStack().pop()).show();
	}
	
}

File: src/lab/miniProject_TS/src/page/addProductPage.ts
import { Page } from "../interface"
import Product from "../product"
import ProductManager from "../productManager"
import State from "../state"

class AddProductPage implements Page {
    
    constructor() {}
    
    public show(): void {
        const products: ProductManager = State.getState().getProductManager()
        products.showProducts()
        const rl = State.getState().getRl()
        let id = State.getState().getId()
        rl.question('Please enter product name: ', (name: string) => {
            rl.question('Please enter product price: ', (price: number) => {
                State.getState().setId(++id)
                let newProduct: Product = new Product(++id, name, price)
                products.addProduct(newProduct)
                State.getState().getStack().pop()?.show()
            })
        })

    }


}

export default AddProductPage
